src/pubsub/mod_pubsub_cache_rdbms.erl

```diff
@@ -0,0 +1,172 @@
+delete_pubsub_last_item(Nidx) ->
+    ["DELETE FROM pubsub_last_item"
+    " WHERE nidx = ", esc_int(Nidx), ";"].
+
+ -spec upsert_pubsub_last_item(
+    Nidx::mod_pubsub:nodeIdx(),
+    ItemId::mod_pubsub:itemId(),
+    Publisher::jid:jid(),
+    Payload::mod_pubsub:payload(),
+    Backend::{atom(), atom()}) -> iolist().
+upsert_pubsub_last_item(Nidx, ItemId, Publisher, Payload, {pgsql, _}) ->
+    upsert_parametrized(Nidx, ItemId, Publisher, Payload, pgsql);
+upsert_pubsub_last_item(Nidx, ItemId, Publisher, Payload, {mysql, _}) ->
+    upsert_parametrized(Nidx, ItemId, Publisher, Payload, mysql);
+upsert_pubsub_last_item(Nidx, ItemId, Publisher, Payload, {odbc, mssql}) ->
+    {
+     EscNidx, EscItemId,
+     EscModifiedLUser, EscModifiedLServer,
+     EscCreatedAt, EscPayload
+    } = esc_query_parms(Nidx, ItemId, Publisher, Payload),
+    PayloadXML = exml:to_binary(Payload),
+    MSSQLPayload = mongoose_rdbms:use_escaped_binary(mongoose_rdbms:escape_binary(global, PayloadXML)),
+    ["MERGE INTO pubsub_last_item with (SERIALIZABLE) as target"
+     " USING (SELECT ", EscNidx, " AS nidx )"
+            " AS source (nidx)"
+        " ON target.nidx = source.nidx"
+     " WHEN MATCHED THEN UPDATE"
+       " SET ",
+        "nidx = ", EscNidx, ", "
+        "itemid = ", EscItemId, ", "
+        "created_luser = ", EscModifiedLUser, ", "
+        "created_lserver = ", EscModifiedLServer, ", "
+        "created_at = ", EscCreatedAt, ", "
+        "payload = ", MSSQLPayload,
+     " WHEN NOT MATCHED THEN INSERT ",
+        columns(),
+         " VALUES (",
+            EscNidx,", ",
+            EscItemId,", ",
+            EscModifiedLUser,", ",
+            EscModifiedLServer,", ",
+            EscCreatedAt,", ",
+            MSSQLPayload,
+          ");"].
+
+upsert_parametrized(Nidx, ItemId, Publisher, Payload, OnConflictLine) ->
+        {
+        EscNidx, EscItemId,
+        EscModifiedLUser, EscModifiedLServer,
+        EscCreatedAt, EscPayload
+    } = esc_query_parms(Nidx, ItemId, Publisher, Payload),
+    [
+        "INSERT INTO pubsub_last_item ",
+        columns(),
+        " VALUES (",
+            EscNidx,", ",
+            EscItemId,", ",
+            EscModifiedLUser,", ",
+            EscModifiedLServer,", ",
+            EscCreatedAt,", ",
+            EscPayload,
+            ") ",
+        on_conflict_line(OnConflictLine),
+            "nidx = ", EscNidx, ", ",
+            "itemid = ", EscItemId, ", ",
+            "created_luser = ", EscModifiedLUser, ", ",
+            "created_lserver = ", EscModifiedLServer, ", ",
+            "created_at = ", EscCreatedAt, ", ",
+            "payload = ", EscPayload,
+            ";"].
+
+
+%%====================================================================
+%% Helpers
+%%====================================================================
+
+check_rdbms_response({selected, [SelectedItem]}) ->
+    LastItem = item_to_record(SelectedItem),
```

Comment:
The purpose of this function is to "check", not "convert". Either this step should be reflected in function name or done in the function that actually does the query (and would be consistent with the way it is done in main DB backend.